src/engine/info-plugin.ts

```diff
@@ -13,6 +13,30 @@ export interface EngineInfo {
 
 export const engineInfo = Symbol()
 
+/**
+ * Extracts the title from the higher hierarchy heading in the Markdown tokens.
+ * @param tokens - The list of Markdown tokens.
+ * @returns The extracted title or undefined if no heading is found.
+ */
+function extractTitleFromFirstHeading(tokens: any[]): string | undefined {
+  let bestHeading: { level: number; content: string } | undefined
+  for (let i = 0; i < tokens.length; i++) {
+    const token = tokens[i]
+    if (token.type === 'heading_open') {
+      const level = parseInt(token.tag.slice(1), 10)
+      const headingContentToken = tokens[i + 1]
+      if (
+        headingContentToken &&
+        headingContentToken.type === 'inline' &&
+        (!bestHeading || level < bestHeading.level)
+      ) {
+        bestHeading = { level, content: headingContentToken.content.trim() }
```

Comment:
For the initial release is fine with this. In future releases, it would be better to use `MarkdownIt.renderInline()` to reflect the exact contents.